src/server/storage/storageAPI.js

```diff
@@ -39,8 +39,14 @@ const setup = (app: Router, storage: StorageAPI) => {
       )
         throw new Error('User email or mobile not verified!')
 
-      const { email, mobile, ...bodyUser } = body.user
-      const user: UserRecord = defaults(bodyUser, {
+      const { email } = body.user
+      const isDuplicatedEmail = await storage.isDupUserData({ email })
```

Comment:
this is going to fail for every new user.
you need to check if createdDate exists, thats the parameter we set when we first add user

Reply:
@sirpy I don't think so. It just checking if there is record with provided email in user collection.

Reply:
user email field was set when user verified his email, so this WILL fail once he calls /user/add.
Just replace check with the createdDate
Stas added a fix that email/mobile will be saved only in otp record and set in /user/add so after his fix this might not fail. But still the correct way is just to check the userrecord date created, if it exists then user can't call /user/add again

Reply:
done

Reply:
done